lib/CHeroHandler.h

```diff
@@ -135,7 +135,7 @@ class DLL_LINKAGE CHero
 
 // convert deprecated format
 std::vector<std::shared_ptr<Bonus>> SpecialtyInfoToBonuses(const SSpecialtyInfo & spec, int sid);
-std::vector<std::shared_ptr<Bonus>> SpecialtyBonusToBonuses(const SSpecialtyBonus & spec);
+std::vector<std::shared_ptr<Bonus>> SpecialtyBonusToBonuses(const SSpecialtyBonus & spec, int sid);
 

```

Comment:
How about default sid value to make this argument optional?

Reply:
optional argument... maybe boost::optional will be useful for this?

Reply:
Nope, I meant just default `sid = 0` so coder doesn't need to write anything to skip this new feature.

Reply:
I could add it, but is there reason to? This function is used for converting deprecated format, so it's not likely to be used more